src/goto-symex/symex_goto.cpp

```diff
@@ -514,10 +523,16 @@ static void merge_names(
     symex_targett::assignment_typet::PHI);
 }
 
-void goto_symext::phi_function(
-  const goto_statet &goto_state,
-  statet &dest_state)
+void goto_symext::phi_function(goto_statet &goto_state, statet &dest_state)
 {
+  // If our destination is unreachable just replace with the incoming state.
+  if(dest_state.guard.is_false())
+  {
+    dest_state.level2 = std::move(goto_state.level2);
+    dest_state.propagation = std::move(goto_state.propagation);
+    return;
+  }
```

Comment:
I'm not sure `phi_function` is the right place for this, I'd do it at the caller (and not invoke `phi_function`).